esp-hal/src/analog/adc/riscv.rs

```diff
@@ -192,6 +191,12 @@ impl super::CalibrationAccess for crate::peripherals::ADC1<'_> {
             AdcCalSource::Gnd => regi2c::ADC_SAR1_ENCAL_GND.write_field(enable as _),
             #[cfg(not(esp32h2))]
             AdcCalSource::Ref => regi2c::ADC_SAR1_ENCAL_REF.write_field(enable as _),
+            // For the ESP32-H2 ground and internal reference voltage are mutually exclusive and
+            // you can toggle between them.
+            //
+            // See: https://github.com/espressif/esp-idf/blob/5c51472e82a58098dda8d40a1c4f250c374fc900/components/hal/esp32h2/include/hal/adc_ll.h#L645
+            #[cfg(esp32h2)]
+            AdcCalSource::Ref => regi2c::ADC_SAR1_ENCAL_GND.write_field(!enable as _),
```

Comment:
I am totally not sure if I understood this correctly, and this might be miles off on how to switch between ground and reference voltage calibration.